File: 0318/318.go
func maximumProductOfWordLengths(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximumProductOfWordLengths(root.Left)
	r := maximumProductOfWordLengths(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 0318/318.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maximumProductOfWordLengths = function(root) {
    if (!root) return 0;
    return 1 + Math.max(maximumProductOfWordLengths(root.left), maximumProductOfWordLengths(root.right));
};